ui/panels-plugin/src/plugins/time-series-chart/utils/palette-gen.ts

```diff
@@ -23,3 +27,26 @@ export function getRandomColor(identifier: string): string {
   const color = `hsla(${~~(180 * hash)},50%,50%,0.8)`;
   return color;
 }
+
+/**
+ * Get line color as well as color for tooltip and legend, account for whether palette is 'Cateogrical' or 'Auto' (generative)
+ */
+export function getSeriesColor(
+  name: string,
+  seriesCount: number,
+  palette: string[],
+  paletteKind: PaletteOptions['kind'] = 'Auto'
+): string {
+  if (paletteKind === 'Auto') {
+    return getRandomColor(name);
```

Comment:
should we just make `Categorical` the default since the current `getRandomColor` doesn't seem the best?

Reply:
We are planning to make `getRandomColor` a lot better, the SO article we took the algorithm from actually has a few improvements we can change to that will increase color contrast: https://stackoverflow.com/questions/3426404/create-a-hexadecimal-colour-based-on-a-string-with-javascript/31037383#31037383

Reply:
Downside of `Categorical` is the colors start to repeat and they aren't consistent across panels. With `Auto` it works better with many time series and the colors will be the same across panels as long as the series names are the same.